### PR TITLE
Add dependabot config

### PR DESCRIPTION
**Pull request by MartinForReal**

### CODE REVIEW

**Comment by MartinForReal** [comment]
@hakman @vteratipally Any ideas? Thanks!

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: 2
+updates:
+  - package-ecosystem: "gomod"
+    directory: "/"
+    open-pull-requests-limit: 1
```
Comment (hakman) [inline .github/dependabot.yml:5]:
```suggestion
    open-pull-requests-limit: 5
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    schedule:
+      interval: weekly
+      time: "01:00"
```
Comment (hakman) [inline .github/dependabot.yml:8]:
```suggestion
      interval: daily
      time: "04:00"
    rebase-strategy: "disabled"
```

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    labels:
+      - "area/dependency"
+      - "ok-to-test"
+      - "lgtm"
+      - "approved"
+
+  - package-ecosystem: "gomod"
+    directory: "/test"
+    open-pull-requests-limit: 1
```
Comment (hakman) [inline .github/dependabot.yml:17]:
```suggestion
    open-pull-requests-limit: 5
```

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    schedule:
+      interval: weekly
+      time: "01:00"
```
Comment (hakman) [inline .github/dependabot.yml:20]:
```suggestion
      interval: daily
      time: "04:00"
    rebase-strategy: "disabled"
```

**Review by hakman — CHANGES_REQUESTED** [review]
I would be comfortable with this with some proposed changes, but I would still leave the `lgtm` label out, so that someone can review and add /lgtm later (anyone in the org can add /lgtm).

**Comment by hakman** [comment]
@vteratipally Any thoughts on this?

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    directory: "/"
+    open-pull-requests-limit: 5
+    schedule:
+      interval: daily
+      time: "04:00"
+    rebase-strategy: "disabled"
+    labels:
+      - "area/dependency"
+      - "ok-to-test"
+      - "lgtm"
```
Comment (hakman) [inline .github/dependabot.yml:13]:
```suggestion
```

**Comment by vteratipally** [comment]
We follow the kubernetes guidelines, let me confirm with other OWNERS.

**Comment by MartinForReal** [comment]
Any update would be appreciated

**Comment by MartinForReal** [comment]
https://www.thoughtworks.com/en-us/radar/techniques/automatic-merging-of-dependency-update-prs @vteratipally That might be something worthy of trying out.

**Comment by MartinForReal** [comment]
@vteratipally @hakman  @mmiranda96 Any advice is appreciated

**Comment by wangzhen127** [comment]
We are updating dependencies weekly now. Do we still need this?

**Comment by hakman** [comment]
@wangzhen127 I think we can close this for now.
It was a good idea and debate around it. Thanks @MartinForReal!

**Pull request closed without merging**